parser/lexer_test.go

```diff
@@ -110,77 +113,79 @@ foo
-		{t: _FLOAT_LIT, line: 27, col: 9, span: 8, v: 123.1234},
-		{t: _FLOAT_LIT, line: 28, col: 9, span: 5, v: 0.123},
-		{t: _INT_LIT, line: 29, col: 9, span: 6, v: uint64(012345)},
-		{t: _INT_LIT, line: 30, col: 9, span: 14, v: uint64(0x2134abcdef30)},
-		{t: '-', line: 31, col: 9, span: 1, v: nil},
-		{t: _INT_LIT, line: 31, col: 10, span: 4, v: uint64(0543)},
-		{t: '-', line: 32, col: 9, span: 1, v: nil},
-		{t: _INT_LIT, line: 32, col: 10, span: 6, v: uint64(0xff76)},
-		{t: _FLOAT_LIT, line: 33, col: 9, span: 8, v: 101.0102},
-		{t: _FLOAT_LIT, line: 34, col: 9, span: 10, v: 202.0203e1},
-		{t: _FLOAT_LIT, line: 35, col: 9, span: 12, v: 304.0304e-10},
-		{t: _FLOAT_LIT, line: 36, col: 9, span: 10, v: 3.1234e+12},
-		{t: '{', line: 38, col: 9, span: 1, v: nil},
-		{t: '}', line: 38, col: 11, span: 1, v: nil},
-		{t: '+', line: 38, col: 13, span: 1, v: nil},
-		{t: '-', line: 38, col: 15, span: 1, v: nil},
-		{t: ',', line: 38, col: 17, span: 1, v: nil},
-		{t: ';', line: 38, col: 19, span: 1, v: nil},
-		{t: '[', line: 40, col: 9, span: 1, v: nil},
-		{t: _OPTION, line: 40, col: 10, span: 6, v: "option"},
-		{t: '=', line: 40, col: 16, span: 1, v: nil},
-		{t: _NAME, line: 40, col: 17, span: 3, v: "foo"},
-		{t: ']', line: 40, col: 20, span: 1, v: nil},
-		{t: _SYNTAX, line: 41, col: 9, span: 6, v: "syntax"},
-		{t: '=', line: 41, col: 16, span: 1, v: nil},
-		{t: _STRING_LIT, line: 41, col: 18, span: 8, v: "proto2"},
-		{t: ';', line: 41, col: 26, span: 1, v: nil},
-		{t: _FLOAT_LIT, line: 44, col: 9, span: 5, v: 1.543, comments: []string{"// some strange cases\n"}},
-		{t: _NAME, line: 44, col: 15, span: 3, v: "g12"},
-		{t: _FLOAT_LIT, line: 45, col: 9, span: 7, v: 0.0, comments: []string{"/* trailing line comment */"}, trailCount: 1},
-		{t: _FLOAT_LIT, line: 46, col: 9, span: 6, v: 0.1234},
-		{t: _FLOAT_LIT, line: 46, col: 16, span: 5, v: 0.5678},
-		{t: '.', line: 46, col: 22, span: 1, v: nil},
-		{t: _FLOAT_LIT, line: 47, col: 9, span: 5, v: 12e12},
-		{t: _FLOAT_LIT, line: 47, col: 15, span: 19, v: math.Inf(1)},
-		{t: _NAME, line: 49, col: 9, span: 53, v: "Random_identifier_with_numbers_0123456789_and_letters"},
-		{t: '.', line: 49, col: 62, span: 1, v: nil},
-		{t: '.', line: 49, col: 63, span: 1, v: nil},
-		{t: '.', line: 49, col: 64, span: 1, v: nil},
-		{t: _NAME, line: 59, col: 9, span: 3, v: "foo", comments: []string{"// this is a trailing comment\n", "// that spans multiple lines\n", "// over two in fact!\n", "/*\n\t * this is a detached comment\n\t * with lots of extra words and stuff...\n\t */", "// this is an attached leading comment\n"}, trailCount: 3},
-		{t: _FLOAT_LIT, line: 61, col: 9, span: 8, v: 1.23e+20},
-		{t: '+', line: 61, col: 17, span: 1, v: nil},
-		{t: _INT_LIT, line: 61, col: 18, span: 2, v: uint64(20)},
+		0:  {t: _INT32, line: 8, col: 9, span: 5, v: "int32", comments: []string{"// comment\n", "/*\n\t * block comment\n\t */", "/* inline comment */"}},
```

Comment:
I added explicit indices just to make it easier to identify which entry fails when there is an error in test output. Sorry for the diff cheese this introduces.